code/go/internal/validator/semantic/validate_field_groups.go

```diff
@@ -31,18 +33,18 @@ type field struct {
 func ValidateFieldGroups(pkgRoot string) ve.ValidationErrors {
```

Comment:
Originally this function (and whole file) was designed to check field group properties. See [comment](https://github.com/elastic/package-spec/blob/cadd81674dc98da1aea3220df313b6085905114f/code/go/internal/validator/semantic/validate_field_groups.go#L32).

Does validation logic of dimensions apply to ordinary, single fields? Maybe it's worth extracting it to a separate .go file?